tests/test_crud.py

```diff
@@ -42,6 +42,8 @@ def test_crud(tmpdir, module, vcrmode):
 def test_check_mode(tmpdir, module):
     if module in ['subscription_manifest', 'templates_import', 'puppetclasses_import', 'content_rhel_role']:
         pytest.skip("This module does not support check_mode.")
+    if module in ['wait_for_task']:
+        pytest.skip("The tests for this module do not support check_mode.")
```

Comment:
This is because we can never create a task to wait on in check_mode? Is it worth supporting check_mode by this module at all?
What if we moved the "deleting a product" task out of the test section and have the running task imprinted in the fixtures already?

Reply:
Maybe this even speeds up the tests altogether.

Reply:
it's because we can't trigger the delete "async" in check mode. (async and check mode are not allowed together in ansible)

so moving the delete to the setup phase? that could work. yeah.